interfaces/kmod/kmod_test.go

```diff
@@ -0,0 +1,101 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package kmod_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/interfaces/backendtest"
+	"github.com/snapcore/snapd/interfaces/kmod"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/testutil"
+	. "gopkg.in/check.v1"
+)
+
+type kmodSuite struct {
+	backendtest.BackendSuite
+}
+
+var _ = Suite(&kmodSuite{})
+
+func (s *kmodSuite) SetUpTest(c *C) {
+	s.Backend = &kmod.Backend{}
+	s.BackendSuite.SetUpTest(c)
+
+	// Prepare a directory for kernel modules conf files.
+	// NOTE: Normally this is a part of the OS snap.
+	err := os.MkdirAll(dirs.SnapKModModulesDir, 0700)
```

Comment:
Again, you should not have to do this in the test setup. The backend must be capable o dealing with this internally.